src/Nfts/Nfts.tsx

```diff
@@ -4,61 +4,72 @@ import {RefreshControl, ScrollView, StyleSheet, Text, View} from 'react-native'
 import {SafeAreaView} from 'react-native-safe-area-context'
 
 import {Icon, Spacer} from '../components'
-import {useFilteredNfts} from './hooks'
+import {useSearch} from '../Search/SearchContext'
+import {useSelectedWallet} from '../SelectedWallet'
+import {useNfts, YoroiNft} from '../yoroi-wallets'
+import {filterNfts} from './filterNfts'
 import {ImageGallery, SkeletonGallery} from './ImageGallery'
 import {useNavigateTo} from './navigation'
 import {NoNftsScreen} from './NoNftsScreen'
 
 export const Nfts = () => {
   const navigateTo = useNavigateTo()
-  const handleNftSelect = (index: number) => navigateTo.nftDetails(nfts[index].id)
   const [isManualRefreshing, setIsManualRefreshing] = React.useState(false)
+  const wallet = useSelectedWallet()
   const strings = useStrings()
 
-  const {search, nfts, isLoading, refetch, isError} = useFilteredNfts({
+  const {isLoading, nfts, refetch, isError} = useNfts(wallet, {
     onSettled: () => {
       if (isManualRefreshing) setIsManualRefreshing(false)
     },
   })
 
+  const {search: nftsSearchTerm} = useSearch()
+  const filteredNfts = filterNfts(nftsSearchTerm, nfts)
+  const sortedNfts = filteredNfts.sort(sortNfts)
+  const nftsSearchResult = filterNfts(nftsSearchTerm, sortedNfts)
+
+  const hasEmptySearchResult = nftsSearchTerm.length > 0 && nftsSearchResult.length === 0
+  const hasNotNfts = nftsSearchTerm.length === 0 && nftsSearchResult.length === 0
+
   const onRefresh = React.useCallback(() => {
     setIsManualRefreshing(true)
     refetch()
   }, [refetch])
 
   if (isError) {
     return (
-      <ScreenWrapper>
+      <Wrapper>
         <ErrorScreen onRefresh={onRefresh} isRefreshing={isManualRefreshing} />
-      </ScreenWrapper>
+      </Wrapper>
     )
   }
 
   if (isLoading) {
     return (
-      <ScreenWrapper>
-        <LoadingScreen nftsCount={nfts.length} />
-      </ScreenWrapper>
+      <Wrapper>
+        <LoadingScreen nftsCount={nftsSearchResult.length} />
+      </Wrapper>
     )
   }
 
-  if (search.length > 0 && nfts.length === 0) {
+  if (hasEmptySearchResult) {
```

Comment:
(isError, isLoading) = 1 scope
(hasNotNfts, hasEmptySearchResults) = different scope
don't mix scopes

Reply:
What would be the implementation that you suggest? Could you specify?

Thanks!

Reply:
I created a ticket to implement the rescoping https://emurgo.atlassian.net/browse/YOMO-518

Thanks for the suggestion.